lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -102,6 +102,11 @@ static CodableConformanceType typeConformsToCodable(DeclContext *context,
 /// \param proto The \c ProtocolDecl to check conformance to.
 static CodableConformanceType
 varConformsToCodable(DeclContext *DC, VarDecl *varDecl, ProtocolDecl *proto) {
+  auto declInterfaceType = varDecl->getValueInterfaceType();
+  if (declInterfaceType->hasError()) {
+    return CodableConformanceType::TypeNotValidated;
```

Comment:
This is weird in my opinion because having error type there means that declaration has been validated and determined to be incorrect. Is that right, @CodaFi?

Reply:
Humm, yes now that you mentioned this seems not correct
But for situations like this
```swift
struct X: Codable { 
   var m: Undefined!
}
```
We can't really return `NotConform` and diagnose `cannot automatically synthesize 'Encodable' because '<<error type>>' does not conform to 'Encodable'`, we could skip validation for error types and not make this call ... of we could skip all derived conformances validation if any of the members have an error type because in that case, we have a problem somewhere else(that should probably be diagnosed already)

Reply:
Maybe skipping validation for error types is the best option here, I am not sure.

Reply:
Great, I think that makes sense so made this change and let's wait for @CodaFi thoughts
Thank you @xedin :)